### PR TITLE
chore: make sui workflow reusable

### PR DESCRIPTION
**Pull request by npty**
# Description

Enhances Sui workflow reusability by changing Sui download path to avoid conflicts with `axelar-contract-deployment` repo, allowing custom `version.json` path, and refactoring for easier integration across projects.

### CODE REVIEW

File: .github/actions/install/action.yaml [hunk 1]
```diff
@@ -28,19 +34,17 @@ runs:
       shell: bash
       if: steps.cache-sui.outputs.cache-hit != 'true'
       run: |
-        curl -L -o sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz https://github.com/MystenLabs/sui/releases/download/${{ env.SUI_VERSION }}/sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz
-        tar -xvf sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz
         mkdir -p sui-binaries
-        mv ./sui ./sui-binaries/
-        mv ./sui-debug ./sui-binaries/
+        curl -L -o sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz https://github.com/MystenLabs/sui/releases/download/${{ env.SUI_VERSION }}/sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz
+        tar -xvf sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz -C sui-binaries
```
Comment (milapsheth) [inline .github/actions/install/action.yaml:39]: :+1:

File: .github/actions/install/action.yaml [hunk 2]
```diff
@@ -28,19 +34,17 @@ runs:
...
         rm -rf sui-${{ env.SUI_VERSION }}-ubuntu-x86_64.tgz
 
     - name: Save Sui binaries
       if: steps.cache-sui.outputs.cache-hit != 'true'
       id: cache-sui-save
       uses: actions/cache@v4
       with:
-        path: sui-binaries/
+        path: download/sui-binaries/
```
Comment (milapsheth) [inline .github/actions/install/action.yaml:47]: why the extra `download/`?

File: version.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-    "SUI_VERSION": "testnet-v1.32.0"
-}
\ No newline at end of file
+    "SUI_VERSION": "mainnet-v1.32.2"
```
Comment (milapsheth) [inline version.json:2]: update all move.toml files

File: .github/actions/install/action.yaml (hunk 2 above)
Comment (npty) [reply .github/actions/install/action.yaml]: good catch! my mistake 🙏🏻

**Pull request merged**